### PR TITLE
Use token based authentication to reset passwords

### PR DESCRIPTION
**Pull request by 0livd**
Send a mail containing a password reset
token link instead of sending a clear text
password.

Ref #232

### CODE REVIEW

**Comment by almet** [comment]
The approach seems good to me, we probably should go ahead with this.

Python 3.5 and python 3.6 seems to be failing, though. From the logs, I believe it's related to tox and an incompatibility with virtualenv. Since this was triggered a while ago, the conflicts might simply be gone, so I triggered another build to see if that makes any difference.

**Comment by 0livd** [comment]
Seems we are not the only ones hitting this issue, a work around is proposed here https://github.com/pytest-dev/pytest/issues/2779#issuecomment-334393688

File: ihatemoney/templates/password_reminder.fr [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Salut,
 
-Vous avez demandez des informations sur votre mot de passe pour "{{ project.name }}".
+Vous avez demandé à réinitialiser votre mot de passe pour "{{ project.name }}". Ce lien est seulement valide pendant30 minutes.
```
Comment (almet) [inline ihatemoney/templates/password_reminder.fr]: missing a space here!

Comment (almet) [reply ihatemoney/templates/password_reminder.fr]: In the french and english version, I believe the `it's only valid for X minutes` should be *after* the actual link.

File: ihatemoney/templates/reminder_mail.en [hunk 2]
```diff
@@ -2,7 +2,7 @@ Hi,
 
 You have just (or someone else using your email address) created the project "{{ g.project.name }}" to share your expenses.
 
-You can access it here: {{ config['SITE_URL'] }}{{ url_for(".list_bills") }} (the identifier is {{ g.project.id }}),
+You can access it here: {{ url_for(".list_bills", _external=True) }} (the identifier is {{ g.project.id }}),
```
Comment (almet) [inline ihatemoney/templates/reminder_mail.en]: Yes, it's better with the use of `_external=True`, thanks!

File: ihatemoney/translations/fr/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -500,6 +532,10 @@ msgstr "Rappel du code d'accès"
 msgid "Your projects"
 msgstr "Vos projets"
 
+#: templates/reset_password.html:2
+msgid "Reset your password"
+msgstr "Changez votre de passe"
```
Comment (JocelynDelalande) [inline ihatemoney/translations/fr/LC_MESSAGES/messages.po]: typo

**Comment by 0livd** [comment]
@almet @JocelynDelalande Just pushed the requested changes. I extended the token lifetime to one hour.

**Comment by almet** [comment]
Okay, since you've addressed all the comments from @JocelynDelalande , I'll take the liberty to merge this :) Thanks again for this, it'll be a lot better now without sending these passwords by email !

**Comment by JocelynDelalande** [comment]
Excellent, thanks to you two!

**Pull request merged**